sos/cleaner/parsers/hostname_parser.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.cleaner.parsers import SoSCleanerParser
+from sos.cleaner.mappings.hostname_map import SoSHostnameMap
+
+
+class SoSHostnameParser(SoSCleanerParser):
+
+    name = 'Hostname Parser'
+    map_file_key = 'hostname_map'
+    prep_map_file = 'sos_commands/host/hostname'
+    regex_patterns = [
+        r'(((\b|_)[a-zA-Z0-9-\.]{1,200}\.[a-zA-Z]{1,63}\b))'
+    ]
+
+    def __init__(self, conf_file=None, opt_domains=None):
+        super(SoSHostnameParser, self).__init__(conf_file)
+        self.mapping = SoSHostnameMap(opt_domains)
+
+    def load_hostname_into_map(self, hostname_string):
+        """Force add the domainname found in /sos_commands/host/hostname into
+        the map. We have to do this here since the normal map prep approach
+        from the parser would be ignored since the system's hostname is not
+        guaranteed
+        """
+        domains = hostname_string.split('.')
+        if len(domains) > 3:
+            # make sure we implicitly get example.com if the system's hostname
+            # is something like foo.bar.example.com
+            high_domain = '.'.join(domain[-2:])
```

Comment:
`s/domain/domains/` in the `join` (currently it raises `name 'domain' is not defined`)

Reply:
Ack